messageCheck/messageVerifier.go

```diff
@@ -140,7 +149,14 @@ func (m *messageVerifier) Serialize(messages []p2p.MessageP2P) ([]byte, error) {
 		pubsubMessages = append(pubsubMessages, pubsubMsgBytes)
 	}
 
-	messagesBytes, err := m.marshaller.Marshal(pubsubMessages)
+	if len(pubsubMessages) == 0 {
+		return []byte{}, nil
```

Comment:
in this case, when returning an empty byte slice because the pubsubMessages is empty, it will still make L171 work? Can you add a test?
Optional, you might return `make([]byte, 0)`